usdhc: mimxrt usdhc implementation under sdhc disk framework.

JunYangNXP opened the pull request.

agansari commented on subsys/disk/disk_access_sdhc.h:
```diff
@@ -0,0 +1,500 @@
+#define SDHC_R1_ADDRESS 0x20
+#define SDHC_R1_PARAMETER 0x40
+
+#define SDHC_SECTOR_SIZE 512
```
Not sure we should have this parameter as a macro, rather set it in Kconfig.
This is the only parameter I've noticed, there may be others.

JunYangNXP replied:
Hi,
http://users.ece.utexas.edu/~valvano/EE345M/SD_Physical_Layer_Spec.pdf 7.3.2.1 Format R1

dleach02 replied:
How is section 7.3.2.1 a response to SDHC_SECTOR_SIZE?

JunYangNXP replied:
define SDHC_SECTOR_SIZE as 512B, which is the smallest sector size of SD. To get better R/W performance, we can set sector size by CMD16 to multiple 512B.

agansari reviewed (changes requested):
Will come back later with more in-depth review

JunYangNXP commented:
Hello Andrei,
These MACROs come from SD response1 type definition, according to the SD specification.

BR
Jun

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Andrei Gansari <notifications@github.com>
发送时间: 星期三, 五月 29, 2019 6:43 下午
收件人: zephyrproject-rtos/zephyr
抄送: Jun Yang; Author
主题: [EXT] Re: [zephyrproject-rtos/zephyr] Usdhc dev b (#16468)


Caution:EXT Email

@agansari requested changes on this pull request.

Will come back later with more in-depth review

________________________________

In subsys/disk/disk_access_sdhc.h<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r288500345&data=02%7C01%7Cjun.yang%40nxp.com%7C91cfe2b3c2b14bf0e13608d6e4227752%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636947234012018058&sdata=1QKJgJP788zWyfSxlB1aW0B7POz3ldiR%2B6NIw8RTrLo%3D&reserved=0>:

> +     SDHC_APP_CLEAR_CARD_DETECT = 42,+       SDHC_APP_SEND_SCR = 51+};++#define SDHC_SEND_OP_COND SDHC_APP_SEND_OP_COND++/* R1 response status */+#define SDHC_R1_IDLE 0x01+#define SDHC_R1_ERASE_RESET 0x02+#define SDHC_R1_ILLEGAL_COMMAND 0x04+#define SDHC_R1_COM_CRC 0x08+#define SDHC_R1_ERASE_SEQ 0x10+#define SDHC_R1_ADDRESS 0x20+#define SDHC_R1_PARAMETER 0x40++#define SDHC_SECTOR_SIZE 512

Not sure we should have this parameter as a macro, rather set it in Kconfig.
This is the only parameter I've noticed, there may be others.

―
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%3Femail_source%3Dnotifications%26email_token%3DALXWABOWQRHB6R26QLGHGYLPXZM4NA5CNFSM4HQKMQCKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZ7IKZI%23pullrequestreview-243172709&data=02%7C01%7Cjun.yang%40nxp.com%7C91cfe2b3c2b14bf0e13608d6e4227752%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636947234012018058&sdata=Y7iGYEhu8ZGDzjOQmEvu4VByTu1qpHke0tYxmsO8DdM%3D&reserved=0>, or mute the thread<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALXWABL4MOMUHZPQ4ZFGRETPXZM4NANCNFSM4HQKMQCA&data=02%7C01%7Cjun.yang%40nxp.com%7C91cfe2b3c2b14bf0e13608d6e4227752%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636947234012028060&sdata=rWpYjgpC5DyzqwTsQgKaQIDnhPzKX%2FV59to4cG%2Bn%2BrI%3D&reserved=0>.

JunYangNXP commented:
@ntavish , Could you please review and test that this doesn’t break SPI SDHC for you? I don’t have the required hardware(nrf52840_blip olimexino_stm32).
I didn't change any code of fs sample.

JunYangNXP commented:
Default sector size is set as 512B, the smallest SD sector size.
To get better R/W performance, we can set sector size by CMD16 to multiple 512B.

BR
Jun

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________
From: David Leach <notifications@github.com>
Sent: Thursday, May 30, 2019 9:17 PM
To: zephyrproject-rtos/zephyr
Cc: Jun Yang; Author
Subject: [EXT] Re: [zephyrproject-rtos/zephyr] usdhc: mimxrt usdhc implementation under sdhc disk framework. (#16468)


Caution: EXT Email

@dleach02 commented on this pull request.

________________________________

In subsys/disk/disk_access_sdhc.h<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r288985359&data=02%7C01%7Cjun.yang%40nxp.com%7Cfe9586f9014142f4898408d6e50129af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636948190486813858&sdata=FKhucOUg1qiUT0x8TXrDm5A%2BZf3y83ZMhVs1uI73B5s%3D&reserved=0>:

> +     SDHC_APP_CLEAR_CARD_DETECT = 42,
+       SDHC_APP_SEND_SCR = 51
+};
+
+#define SDHC_SEND_OP_COND SDHC_APP_SEND_OP_COND
+
+/* R1 response status */
+#define SDHC_R1_IDLE 0x01
+#define SDHC_R1_ERASE_RESET 0x02
+#define SDHC_R1_ILLEGAL_COMMAND 0x04
+#define SDHC_R1_COM_CRC 0x08
+#define SDHC_R1_ERASE_SEQ 0x10
+#define SDHC_R1_ADDRESS 0x20
+#define SDHC_R1_PARAMETER 0x40
+
+#define SDHC_SECTOR_SIZE 512


How is section 7.3.2.1 a response to SDHC_SECTOR_SIZE?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%3Femail_source%3Dnotifications%26email_token%3DALXWABIYXFN2HDPGQ5VMVETPX7HWNA5CNFSM4HQKMQCKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2D6VEY%23discussion_r288985359&data=02%7C01%7Cjun.yang%40nxp.com%7Cfe9586f9014142f4898408d6e50129af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636948190486813858&sdata=amBJqwfEWkKPIsfxWxhUQ%2BlTqgtLIRNAjjdja3VoZbg%3D&reserved=0>, or mute the thread<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALXWABNMR2A5AYXRWMISUCLPX7HWNANCNFSM4HQKMQCA&data=02%7C01%7Cjun.yang%40nxp.com%7Cfe9586f9014142f4898408d6e50129af%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636948190486813858&sdata=5D4wI2t5Bxh%2BTUZSf8tFSBCCbO68N8W4Qsqx%2BHnf49w%3D&reserved=0>.

ntavish reviewed (approved):
Doesn't break the `subsys/fs` sample.

ydamigos commented on dts/bindings/sdhc/sdhc.yaml:
```diff
@@ -0,0 +1,38 @@
+#
```
Is the goal of this file to collect properties which are common to multiple MMC host controllers?
If yes, I believe that it should be moved under `dts/bindings/mmc` and rename to `mmc.yaml`.

JunYangNXP replied:
moved to dts/bindings/mmc/ and rename to mmc-sdhc.yaml.

JunYangNXP replied:
fixed in commit 9dc0d12bfe05f783471e35a3034e4b9276c8e818

JunYangNXP replied:
fixed in commit 6f1e1579db1ecb4ba8cd27c1c0b13b4ae5d5daf0
nxp,imx-usdhc.yaml: specifies nxp usdhc module.
nxp,imx-usdhc-slot.yaml: specifies usdhc slot on board which inherits usdhc.
on board dts:
&usdhc1 {
       status = "ok";
       compatible = "nxp,imx-usdhc-slot";
       pwr-gpios = <&gpio1 5 0>;
       cd-gpios = <&gpio2 28 0>;
};

ydamigos reviewed (changes requested):
Tested `samples/subsys/fs` on olimexino_stm32 and it works fine.
A couple of initial comments but It will take some time to review it.

JunYangNXP commented:
Hello Yannis,
Really appreciate your great support for my PR.
I am looking into the issues you mentioned, and will fix them and push again.

BR
Jun

From: Yannis Damigos [mailto:notifications@github.com]
Sent: Sunday, June 02, 2019 12:08 AM
To: zephyrproject-rtos/zephyr <zephyr@noreply.github.com>
Cc: Jun Yang <jun.yang@nxp.com>; Author <author@noreply.github.com>
Subject: [EXT] Re: [zephyrproject-rtos/zephyr] usdhc: mimxrt usdhc implementation under sdhc disk framework. (#16468)


Caution: EXT Email

@ydamigos requested changes on this pull request.

Tested samples/subsys/fs on olimexino_stm32 and it works fine.
A couple of initial comments but It will take some time to review it.

________________________________

In dts/bindings/sdhc/fsl,imxrt-usdhc.yaml<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609201&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619702531&sdata=czxSqAdmoq84eyE%2BMzXBEG%2BpEo1XLOOBRbmzT9hW3p8%3D&reserved=0>:

> @@ -0,0 +1,31 @@

+#

+# Copyright (c) 2019, NXP

+#

+# SPDX-License-Identifier: Apache-2.0

+#

+---

+title: NXP USDHC MODULE

Shouldn't it be Freescale here?

________________________________

In dts/arm/nxp/nxp_rt.dtsi<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609210&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619702531&sdata=6p3uXZNpbmhz6L2pui05WAbJJVlqFy53h301qUrUWog%3D&reserved=0>:

> @@ -346,6 +346,22 @@

                       status = "disabled";

                       label = "USBD_1";

               };

+

+              usdhc1: usdhc@402c0000 {

+                      compatible = "nxp,imx-usdhc";

"fsl,imxrt-usdhc"?

________________________________

In dts/arm/nxp/nxp_rt.dtsi<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609216&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619712526&sdata=LkNqnWoJuSnWvEcaWLOhb0a5%2B5U4jllinQohzh5w5sk%3D&reserved=0>:

> @@ -346,6 +346,22 @@

                       status = "disabled";

                       label = "USBD_1";

               };

+

+              usdhc1: usdhc@402c0000 {

+                      compatible = "nxp,imx-usdhc";

+                      reg = <0x402c0000 0x4000>;

+                      status = "ok";

+                      interrupts = <110 0>;

+                      label = "USDHC_1";

+              };

+

+              usdhc2: usdhc@402c4000 {

+                      compatible = "nxp,imx-usdhc";

"fsl,imxrt-usdhc"?

________________________________

In dts/bindings/sdhc/fsl,imxrt-usdhc.yaml<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609264&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619712526&sdata=GMA7Zmbp%2FS%2BdpA8kaTc8c3E%2FsiaYntunH1TRThj1d%2BE%3D&reserved=0>:

> +#

+# SPDX-License-Identifier: Apache-2.0

+#

+---

+title: NXP USDHC MODULE

+version: 0.1

+

+description: >

+    This binding gives a base representation of the NXP i.MX RT USDHC module

+

+properties:

+    compatible:

+      type: string

+      category: required

+      description: compatible strings

+      constraint: "nxp,imx-usdhc"

"fsl,imxrt-usdhc"?

________________________________

In dts/arm/nxp/nxp_rt.dtsi<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609536&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619722519&sdata=d7fhRseDmFT7HZ9NeSeMWuDoWgwl%2Ff4pmANFuYwuSWM%3D&reserved=0>:

> @@ -346,6 +346,22 @@

                       status = "disabled";

                       label = "USBD_1";

               };

+

+              usdhc1: usdhc@402c0000 {

+                      compatible = "nxp,imx-usdhc";

+                      reg = <0x402c0000 0x4000>;

+                      status = "ok";

Better "disabled" here and enable it at board level.

________________________________

In dts/arm/nxp/nxp_rt.dtsi<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609539&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619722519&sdata=I4J8QzlO9hb9O2iacLwcc6whOrytRBnDXzGkLzFVZvc%3D&reserved=0>:

> @@ -346,6 +346,22 @@

                       status = "disabled";

                       label = "USBD_1";

               };

+

+              usdhc1: usdhc@402c0000 {

+                      compatible = "nxp,imx-usdhc";

+                      reg = <0x402c0000 0x4000>;

+                      status = "ok";

+                      interrupts = <110 0>;

+                      label = "USDHC_1";

+              };

+

+              usdhc2: usdhc@402c4000 {

+                      compatible = "nxp,imx-usdhc";

+                      reg = <0x402c4000 0x4000>;

+                      status = "ok";

Better "disabled" here and enable it at board level.

________________________________

In boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609737&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619722519&sdata=vNBzGDDXkM192iBdTar0lQ2tuTjaSUjcAE%2Btfj%2FJQeI%3D&reserved=0>:

> @@ -63,6 +63,13 @@

                       };

               };

        };

+

+       sdhc {

I am not sure about this. I was expecting something like:

&usdhc1 {

               status = "ok";

               cd-gpios = <&gpio2 28 0>;

               pwr-gpios = <&gpio1 5 0>;

};

________________________________

In dts/bindings/sdhc/fsl,imxrt-usdhc.yaml<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289609770&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619732515&sdata=GL3YJBNb%2BBlu8nJ871cV%2B6EBpiPS5bThvsdWOR8S2X8%3D&reserved=0>:

> @@ -0,0 +1,31 @@

+#

+# Copyright (c) 2019, NXP

+#

+# SPDX-License-Identifier: Apache-2.0

+#

+---

+title: NXP USDHC MODULE

+version: 0.1

+

+description: >

+    This binding gives a base representation of the NXP i.MX RT USDHC module

Freescale instead of NXP?

________________________________

In dts/bindings/sdhc/fsl,imxrt-usdhc.yaml<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289610010&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619732515&sdata=0b7Q7Y8a0gn78UFD4z5kon5UF%2FZV9gOUShteUDD6pJY%3D&reserved=0>:

> @@ -0,0 +1,31 @@

+#

Should we move this file under dts/bindings/mmc?

________________________________

In dts/bindings/sdhc/sdhc.yaml<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%23discussion_r289610328&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619742507&sdata=7wEy2%2F%2BHGf3VeHyIiS38%2B5H6d53IzqenMiJUKVh%2Bo2g%3D&reserved=0>:

> @@ -0,0 +1,38 @@

+#

Is the goal of this file to collect properties which are common to multiple MMC host controllers?
If yes, I believe that it should be moved under dts/bindings/mmc and rename to mmc.yaml.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fzephyrproject-rtos%2Fzephyr%2Fpull%2F16468%3Femail_source%3Dnotifications%26email_token%3DALXWABILG3BTWUM5RMPV4L3PYKNEXA5CNFSM4HQKMQCKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2KCW3Y%23pullrequestreview-244591471&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619742507&sdata=ziRXhvjks7G4kNPSuBFjjRPCdYOCWPqT7KWyIIOlJOA%3D&reserved=0>, or mute the thread<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALXWABOWIWW2ASEGYLBKOX3PYKNEXANCNFSM4HQKMQCA&data=02%7C01%7Cjun.yang%40nxp.com%7Cfba6136e5f8d4e4c3b6d08d6e6ab45cc%7C686ea1d3bc2b4c6fa92cd99c5c301635%7C0%7C0%7C636950020619742507&sdata=3yeetFnbwLDMU5vfdXgtecbT2gtkPSN6DuoB%2BnCAAlU%3D&reserved=0>.

JunYangNXP commented on dts/bindings/sdhc/fsl,imxrt-usdhc.yaml:
```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: NXP USDHC MODULE
```
move dts/bindings/sdhc/fsl,imxrt-usdhc.yaml to dts/bindings/mmc/nxp,imxrt-usdhc.yaml

JunYangNXP replied:
@ydamigos Could you help review and check this branch with olimexino_stm32 again?
There is code refactoring.

ydamigos replied:
@JunYangNXP I'll try to find some time this weekend to test it.

JunYangNXP commented on dts/arm/nxp/nxp_rt.dtsi:
```diff
@@ -346,6 +346,22 @@
+		};
+
+		usdhc2: usdhc@402c4000 {
+			compatible = "nxp,imx-usdhc";
```
fixed.

JunYangNXP commented on dts/bindings/sdhc/fsl,imxrt-usdhc.yaml:
```diff
@@ -0,0 +1,31 @@
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "nxp,imx-usdhc"
```
fixed.

JunYangNXP commented on dts/arm/nxp/nxp_rt.dtsi:
```diff
@@ -346,6 +346,22 @@
+		usdhc1: usdhc@402c0000 {
+			compatible = "nxp,imx-usdhc";
+			reg = <0x402c0000 0x4000>;
+			status = "ok";
```
fixed.

JunYangNXP commented on dts/arm/nxp/nxp_rt.dtsi:
```diff
@@ -346,6 +346,22 @@
+		usdhc2: usdhc@402c4000 {
+			compatible = "nxp,imx-usdhc";
+			reg = <0x402c4000 0x4000>;
+			status = "ok";
```
fixed.

JunYangNXP commented on boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts:
```diff
@@ -63,6 +63,13 @@
 		};
 	};
+
+	sdhc {
```
Fixed to:
&usdhc1 {
        status = "ok";
        sd-slot {
                compatible = "zephyr,mmc-sd-slot";
                pwr-gpios = <&gpio1 5 0>;
                cd-gpios = <&gpio2 28 0>;
        };
};

ydamigos replied:
Why do we need the sd-slot node? Isn't the usdhc1 node enough? I checked linux dts for similar controllers and it uses only esdhc/usdhc nodes (https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/mmc/fsl-imx-esdhc.txt).

JunYangNXP replied:
Per my understanding, usdhc1 is used to specify the usdhc1 IP of i.MXRT SoC. 
sd-slot is used to specify the slot on board, includes pwr and cd gpios other than sdhc bus(cmd, clk and data connected).

ydamigos replied:
I am not quite sure about sd-slot. Take for example imx6sl-evk on linux.
usdhc1 IP is specified in imx6sl.dtsi:
```
usdhc1: usdhc@2190000 {
		compatible = "fsl,imx6sl-usdhc", "fsl,imx6q-usdhc";
		reg = <0x02190000 0x4000>;
		interrupts = <0 22 IRQ_TYPE_LEVEL_HIGH>;
		clocks = <&clks IMX6SL_CLK_USDHC1>,
			 <&clks IMX6SL_CLK_USDHC1>,
			 <&clks IMX6SL_CLK_USDHC1>;
		clock-names = "ipg", "ahb", "per";
		bus-width = <4>;
		status = "disabled";
};
``` 
cd-gpios and pinctrls are specified in imx6sl-evk.dts
```
&usdhc1 {
	pinctrl-names = "default", "state_100mhz", "state_200mhz";
	pinctrl-0 = <&pinctrl_usdhc1>;
	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
	bus-width = <8>;
	cd-gpios = <&gpio4 7 GPIO_ACTIVE_LOW>;
	wp-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
	status = "okay";
};
```

JunYangNXP replied:
fixed in commit 6f1e1579db1ecb4ba8cd27c1c0b13b4ae5d5daf0
nxp,imx-usdhc.yaml: specifies nxp usdhc module.
nxp,imx-usdhc-slot.yaml: specifies usdhc slot on board which inherits usdhc.
on board dts:
&usdhc1 {
       status = "ok";
       compatible = "nxp,imx-usdhc-slot";
       pwr-gpios = <&gpio1 5 0>;
       cd-gpios = <&gpio2 28 0>;
};

JunYangNXP commented on dts/bindings/sdhc/fsl,imxrt-usdhc.yaml:
```diff
@@ -0,0 +1,31 @@
+version: 0.1
+
+description: >
+    This binding gives a base representation of the NXP i.MX RT USDHC module
```
renamed to nxp,imx-usdhc.yaml

JunYangNXP commented:
After I fix the conflict of "nxp_rt.dtsi". I can't build samples/subsys/fs:

samples/subsys/fs/src/main.c:14:10: fatal error: ff.h: No such file or directory
#include <ff.h>

Anyone help check?

MaureenHelm reviewed (changes requested):
Please rebase your branch onto master and force push, instead of merging master into your branch. Also, when reviewers request changes, you should amend the relevant commit(s) and force push instead of creating new commits. This reduces noise in the upstream git history when the PR gets accepted.

> samples/subsys/fs/src/main.c:14:10: fatal error: ff.h: No such file or directory

fatfs was moved out of the main repo into a module in commit 45868a2dfd65ebaf277222513264b010a4bfeb7b. You need to run `west update`

JunYangNXP commented on boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts:
```diff
@@ -63,6 +63,13 @@
 	};
+
+	sdhc {
+		compatible = "sdhc";
```
fixed in commit 9dc0d12bfe05f783471e35a3034e4b9276c8e818
&usdhc1 {
	status = "ok";
	slot {
		compatible = "zephyr,mmc-sdhc-slot";
		pwr-gpios = <&gpio1 5 0>;
		cd-gpios = <&gpio2 28 0>;
	};
};

JunYangNXP replied:
fixed in commit 6f1e1579db1ecb4ba8cd27c1c0b13b4ae5d5daf0
nxp,imx-usdhc.yaml: specifies nxp usdhc module.
nxp,imx-usdhc-slot.yaml: specifies usdhc slot on board which inherits usdhc.
on board dts:
&usdhc1 {
       status = "ok";
       compatible = "nxp,imx-usdhc-slot";
       pwr-gpios = <&gpio1 5 0>;
       cd-gpios = <&gpio2 28 0>;
};

JunYangNXP commented on boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts:
```diff
@@ -115,6 +116,13 @@ arduino_serial: &uart3 {};
 };
 
+/*spi1 unused, in order to pass compiling fs sample code on top of usdhc
+ * which depends on spi.
```
fixed in commit c191c3e25e0ca4c87b66d564833b9e5f0559768b with additional configure file: prj_mimxrt1050_evk.conf

JunYangNXP replied:
fixed in commit 522089419161b8b1fdc1ac333e8d62fe8cbf091e with additional configure file: prj_mimxrt1050_evk.conf

JunYangNXP commented on subsys/disk/Kconfig:
```diff
@@ -94,11 +94,24 @@ config DISK_VOLUME_SIZE
+if DISK_ACCESS_SDHC
+
+config FSL_USDHC
+	int "Freescale USDHC enable"
```
fixed in commit 8710a847dfdcfa6dcd7712bc302f6173a0197deb

JunYangNXP replied:
fixed in commit 39922dd7737462d9d2ca687091a68f033a1846ad

JunYangNXP commented on soc/arm/nxp_imx/rt/dts_fixup.h:
```diff
@@ -127,4 +127,11 @@
+#define DT_ZEPHYR_MMC_SPI_SLOT_0_BUS_NAME NULL
+#define DT_ZEPHYR_MMC_SPI_SLOT_0_CS_GPIO_CONTROLLER NULL
+#define DT_ZEPHYR_MMC_SPI_SLOT_0_BASE_ADDRESS 0
+#define DT_ZEPHYR_MMC_SPI_SLOT_0_CS_GPIO_PIN 0
```
removed these in commit 9dc0d12bfe05f783471e35a3034e4b9276c8e818

JunYangNXP commented on soc/arm/nxp_imx/rt/Kconfig.soc:
```diff
@@ -81,6 +81,7 @@ config SOC_MIMXRT1052
 	select INIT_USB1_PLL
 	select INIT_VIDEO_PLL if DISPLAY_MCUX_ELCDIF
 	select HAS_MCUX_USB_EHCI
+	select HAS_MCUX_USDHC
```
fixed in commit 0ef5d7c974edb86e709353813703370eef829e6f

JunYangNXP commented on subsys/disk/disk_access_sdhc.c:
```diff
@@ -276,6 +276,10 @@ static bool sdhc_retry_ok(struct sdhc_retry *retry)
 {
+	if (!data->spi) {
+		return;
+	}
```
Fixed. Separate USDHC and SPI SDHC to 2 files.
disk_access_sdhc_spi.c and disk_access_usdhc.c
commit 8a761b3870b5f0129e7643f5267218abe828f5ee

ydamigos commented on dts/bindings/mmc/nxp,imx-usdhc-slot.yaml:
```diff
@@ -0,0 +1,35 @@
+version: 0.1
+
+description: >
+    NXP i.MX USDHC slot
```
I would propose to use a generic mmc.yaml which will contains the properties like compatible, address_cells, size_cells, label, which will be common for different mmc/sdhc controllers. nxp,imx-usdhc.yaml should inherit mmc.yaml and contain properties like reg, interrupts, pwr-gpios, cd-gpios and compatible's constrain.

JunYangNXP replied:
I agree to use generic mmc.yaml which nxp,imx-usdhc.yaml inherits.
But pwr-gpios are cd-gpios are not part of usdhc IP but part of board design.
So we  may have mmc.yaml->nxp,imx-usdhc.yaml(SoC)->nxp,imx-usdhc-slot.yaml(board).
Is it OK?

agansari replied:
is there anythong particular to nxp,imx-usdhc.yaml ? if not, you can skip and just do mmc.yaml->nxp,imx-usdhc-slot.yaml(board)

JunYangNXP replied:
OK, I ill skip nxp,imx-usdhc.yaml

ydamigos replied:
Better to keep as close to linux dts files as possible. We could add cd-gpios and pwr-gpios as optional properties in mmc.yaml. Something similar is done in linux mmc dts binding regarding cd-gpios. Below is a part of linux mmc dts binding documentation:
```
Card detection:
If no property below is supplied, host native card detect is used.
Only one of the properties in this section should be supplied:
  - broken-cd: There is no card detection available; polling must be used.
  - cd-gpios: Specify GPIOs for card detection, see gpio binding
  - non-removable: non-removable slot (like eMMC); assume always present.
```

JunYangNXP replied:
OK, I will use generic mmc.yaml to specify sdhc/mmc and add gpios as optional.

JunYangNXP replied:
Done

agansari commented on dts/bindings/mmc/nxp,imx-usdhc-slot.yaml:
```diff
@@ -0,0 +1,35 @@
+    NXP i.MX USDHC slot
+
+inherits:
+    !include nxp,imx-usdhc.yaml
```
I understand nxp,imx-usdhc-slot.yaml describes hw pins, while nxp,imx-usdhc.yaml describes how the mcu handles sd signals. Why not have a single yaml for all pins, interrupts, or any other configuration.
I would rather have a yaml split between generic mmc/sd and particular nxp configurations. In your current pull i don't see anything that would be nxp particular.

JunYangNXP replied:
according to your suggestion, I use generic mmc.yaml for mmc/sdhc.

agansari commented on subsys/disk/disk_access_sdhc.h:
```diff
@@ -0,0 +1,751 @@
+/* Clock speed used during initialisation */
+#define SDHC_INITIAL_SPEED 400000
+/* Clock speed used after initialisation */
+#define SDHC_SPEED 4000000
```
SDHC_SPEED should be moved to kconfig or dts.

Or have a bus speed defined and do the math from there to determine speed? What do you think?

JunYangNXP replied:
SDHC_INITIAL_SPEED and SDHC_SPEED are only used for spi sdhc to configure spi speed.
I will rename it to avoid confusing and move it from generic header to disk_access_sdhc_spi.c.

JunYangNXP replied:
Done

agansari commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -0,0 +1,2879 @@
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```
I would stay away from NXP particular device configuration in a generic usdhc acces driver.

JunYangNXP replied:
usdhc is just sdhc controller of NXP SoC, not generic.

agansari commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -0,0 +1,2879 @@
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```
You can tie DT_NXP_IMX_USDHC_SLOT_0_LABEL to a generic DT_USDHC_SLOT_0_LABEL using dts_fixup.h

JunYangNXP replied:
usdhc is not generic, only sdhc controller of NXP SoC.

agansari commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -0,0 +1,2879 @@
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```
You can tie DT_NXP_IMX_USDHC_SLOT_0_LABEL to a generic DT_USDHC_SLOT_0_LABEL using dts_fixup.h

JunYangNXP replied:
usdhc is not generic, only sdhc controller of NXP SoC.

agansari commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -0,0 +1,2879 @@
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```
You can tie DT_NXP_IMX_USDHC_SLOT_0_BASE_ADDRESS to a generic DT_USDHC_SLOT_0_BASE_ADDRESS using dts_fixup.h 
This will help drop nxp ifdefs.

JunYangNXP replied:
Hi, I can't fully understand how to drop nxp ifdefs..

agansari replied:
@JunYangNXP See zephyr/soc/arm/nxp_imx/rt/dts_fixup.h
Look at the last definition:
#define DT_USBD_MCUX_EHCI_MAXIMUM_SPEED   DT_NXP_KINETIS_USBD_402E0000_MAXIMUM_SPEED

DT_USBD_MCUX_EHCI_MAXIMUM_SPEED - used in code
DT_NXP_KINETIS_USBD_402E0000_MAXIMUM_SPEED - generated from dts. Look at <sample>/build/zephyr/include/generated/generated_dts_board.conf after compiling at the generated macros from dts.

In your case DT_NXP_IMX_USDHC_SLOT_0_BASE_ADDRESS is a generated macro.
Using a generic definition will help reuse the same code without dts hard-coding.

JunYangNXP replied:
For MIMXRT1050EVK board, the generated macro is:
#define DT_NXP_IMX_USDHC_0_BASE_ADDRESS				DT_NXP_IMX_USDHC_402C0000_BASE_ADDRESS
 but the DT_NXP_IMX_USDHC_0_BASE_ADDRESS depends on board design, if customer uses USDHC2 for their board SDHC slot, the generated macro should be:
#define DT_NXP_IMX_USDHC_0_BASE_ADDRESS				DT_NXP_IMX_USDHC_402C4000_BASE_ADDRESS
seems hard to fix this marco.

please correct me if any mistake.

agansari replied:
The customer will have something like:
#define DT_NXP_IMX_USDHC_>>1<<_BASE_ADDRESS DT_NXP_IMX_USDHC_402C4000_BASE_ADDRESS
This can be added currently in the code.
A custom board would use the same soc configuration, but he would have a new board directory with different pinmux and other cofigurations.

JunYangNXP replied:
DT_NXP_IMX_USDHC_0_BASE_ADDRESS means the SDHC slot 0 (1st) on board, the slot 0 could connect to USDHC0 or USDHC1 which depends on board design.
If USDHC0 is connected, DT_NXP_IMX_USDHC_0_BASE_ADDRESS  is DT_NXP_IMX_USDHC_402C0000_BASE_ADDRESS, otherwise, it's DT_NXP_IMX_USDHC_402C4000_BASE_ADDRESS.

agansari replied:
DT_NXP_IMX_USDHC_x_BASE_ADDRESS refers to SoC internal device's address, not the external slot on the board.
We use dts to associate slot to soc's device.

JunYangNXP replied:
on mimxrt1050_evk.dts:
I defined:
&usdhc1 {
	status = "ok";
	pwr-gpios = <&gpio1 5 0>;
	cd-gpios = <&gpio2 28 0>;
};
And generate macros:
#define DT_NXP_IMX_USDHC_0_BASE_ADDRESS				DT_NXP_IMX_USDHC_402C0000_BASE_ADDRESS

But if I change the board dst:
&usdhc2 {
	status = "ok";
	pwr-gpios = <&gpio1 5 0>;
	cd-gpios = <&gpio2 28 0>;
};
it generates:
#define DT_NXP_IMX_USDHC_0_BASE_ADDRESS				DT_NXP_IMX_USDHC_402C4000_BASE_ADDRESS

agansari replied:
You are right, unless both devices have status = "ok", the only device enabled will have 0 set.

You can have multiple dts_fixup.h some in soc, some in boards. At a board level you can have USDHC_0 defined as using &usdhc1 and another board to use &usdhc2.

Just a few boards have dts_fixup.h defined in board folder, as most development boards for a soc family reuse the same board design.

agansari replied:
Are there other i.MX RT 10xx using a diferent usdhc device for the slot?

JunYangNXP replied:
I did't check other boards, but I think board designer prefer to use USDHC0 unless pins of USDHC0 are muxed for some other purpose.:)

agansari commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -0,0 +1,2879 @@
+
+		ret = usdhc_xfer(priv);
+		if (!ret) {
+			ret = usdhc_
```
if you use generic pin definitons in dts, you can drop nxp specific definitions, same for slot 0 above

JunYangNXP commented on boards/arm/mimxrt1050_evk/pinmux.c:
```diff
@@ -213,6 +299,10 @@ static int mimxrt1050_evk_init(struct device *dev)
 #endif
 
+#ifdef CONFIG_DISK_ACCESS_NXP_USDHC1
+	usdhc_pin_config(0, true, 2, 1);
```
SoC has 2 USDHC1 and USDHC2, but mimxrt1050_evk board only uses one.
So board dts only has definition of USDHC1.

agansari replied:
Understood

JunYangNXP commented on CODEOWNERS:
```diff
@@ -309,6 +309,9 @@
 /subsys/bluetooth/controller/             @carlescufi @cvinayak @thoh-ot
 /subsys/debug/                            @nashif
+/subsys/disk/disk_access_sdhc.h           jun.yang@nxp.com
+/subsys/disk/disk_access_sdhc_spi.c       jun.yang@nxp.com
```
Done

ydamigos commented on dts/bindings/mmc/mmc.yaml:
```diff
@@ -0,0 +1,56 @@
+      category: required
+      description: compatible strings
+      generation: define
+      constraint: "zephyr,mmc-sdhc"
```
We should not use a constraint in the generic mmc.yaml. This should defined in devices specific yaml file (e.g. nxp,imx-usdhc.yaml in your case). You should also take into account PR #16693 which adds a common base.yaml file.

JunYangNXP replied:
I agree. Please check the latest code pushed.

MaureenHelm commented on subsys/disk/Kconfig:
```diff
@@ -110,6 +110,21 @@ config DISK_ACCESS_SPI_SDHC
 	help
 	  File system on a SDHC card accessed over SPI.
 
+config DISK_ACCESS_NXP_USDHC1
```
Make this:
```
config DISK_ACCESS_USDHC
	bool "NXP i.MXRT USDHC driver"
	depends on HAS_MCUX_USDHC
	help
	  Enable support for the i.MXRT USDHC driver

config DISK_ACCESS_USDHC1
	bool "NXP i.MXRT USDHC instance 1"
	depends on HAS_MCUX_USDHC
	help
	  Enable support for the i.MXRT USDHC driver instance 1

config DISK_ACCESS_USDHC2
	bool "NXP i.MXRT USDHC instance 2"
	depends on HAS_MCUX_USDHC
	help
	  Enable support for the i.MXRT USDHC driver instance 2
```

This will allow you to make the `subsys/disk/CMakeLists.txt` look like this:
```
zephyr_sources_ifdef(CONFIG_DISK_ACCESS_USDHC disk_access_usdhc.c)
```

MaureenHelm commented on ext/hal/nxp/mcux/Kconfig:
```diff
@@ -108,4 +108,14 @@ config HAS_MCUX_USB_EHCI
+config HAS_MCUX_USDHC2
+	bool
+	help
+	  Set if the USDHC2 instance 2 module is present in the SoC.
```
Make this:
```
config HAS_MCUX_USDHC
	bool
	help
	  Set if the USDHC module is present in the SoC.
```

JunYangNXP replied:
I am still using "HAS_MCUX_USDHC1" and "HAS_MCUX_USDHC2", Some SoCs are supposed to have only one USDHC instance.
In subsys/disk/Kconfig, the dependency of DISK_ACCESS_USDHC is
"HAS_MCUX_USDHC1" or "HAS_MCUX_USDHC2" like this:
config DISK_ACCESS_USDHC
        bool "NXP i.MXRT USDHC driver"
        depends on (HAS_MCUX_USDHC1 || HAS_MCUX_USDHC2)

MaureenHelm commented on samples/subsys/fs/prj_mimxrt1050_evk.conf:
```diff
@@ -0,0 +1,11 @@
+CONFIG_SPI=y
+CONFIG_SPI_1=n
```
It shouldn't be necessary to add the SPI configs

MaureenHelm commented on subsys/disk/disk_access_sdhc_spi.c:
```diff
@@ -828,7 +827,6 @@ static int disk_spi_sdhc_init(struct device *dev)
 
 static struct sdhc_spi_data sdhc_spi_data_0;
 
-#ifdef DT_ZEPHYR_MMC_SPI_SLOT_0_LABEL
```
Is this right? It seems strange to remove an `#ifdef` without also removing an `#endif`

JunYangNXP replied:
Actually I moved this to the beginning of the file, otherwise, there will be lots of compile warnings in case that no SPI is associated to SDHC but SPI SDHC is configured.
#ifndef DT_INST_0_ZEPHYR_MMC_SPI_SLOT_LABEL
#warning NO SDHC slot specified on board
#else
....
#endif

MaureenHelm commented on soc/arm/nxp_imx/rt/soc.c:
```diff
@@ -208,22 +208,6 @@ static ALWAYS_INLINE void clkInit(void)
-				(CLOCK_GetDiv(kCLOCK_Usdhc1Div) + 1U);
-	} else if (nusdhc == 0) {
-		return CLOCK_GetSysPfdFreq(kCLOCK_Pfd0) /
-				(CLOCK_GetDiv(kCLOCK_Usdhc2Div) + 1U);
```
You added this in a previous commit in the same PR, then removed it. You're right that this code should be in the clock_control driver, but please fix up the git history in the PR

MaureenHelm reviewed (changes requested):
@galak please review the dts

ydamigos commented on dts/arm/nxp/nxp_rt.dtsi:
```diff
@@ -357,6 +357,28 @@
+			clocks = <&ccm IMX_CCM_USDHC1_CLK 0 0>;
+			label = "USDHC_1";
+			#address-cells = <1>;
+			#size-cells = <0>;
```
Properties `#address-cells` and `#size-cells` should be removed as usdhc nodes don't have child nodes (subnodes).

JunYangNXP replied:
fixed.

ydamigos replied:
You forgot to handle `#address-cells` and `#address-cells` in `mmc.yaml`. It should be optional, or better, we could remove them as they aren't used (we could add them later if a new node needs them).

JunYangNXP replied:
I removed them from mmc.yaml

JunYangNXP commented:
PROJECT EXECUTION FAILED
---------out-3nd-pass/qemu_xtensa/tests/kernel/sched/schedule_api/kernel.sched/handler.log----------
6/8 qemu_xtensa               tests/kernel/sched/schedule_api/kernel.sched.multiq PASSED (qemu 8.627s)
7/8 qemu_x86_64               tests/kernel/sched/schedule_api/kernel.sched.multiq PASSED (qemu 19.145s)
8/8 qemu_xtensa               tests/kernel/sched/preempt/kernel.sched.preempt    FAILED: timeout (qemu 60.059s)

The error report by Shippable seems not be associated to this PR.

MaureenHelm commented on dts/bindings/mmc/nxp,imx-usdhc.yaml:
```diff
@@ -0,0 +1,37 @@
+      type: compound
+      category: optional
+      description: Detect pin
+      generation: define, use-prop-name
```
Remove both instances of `use-prop-name`

JunYangNXP replied:
fixed

MaureenHelm commented on soc/arm/nxp_imx/rt/soc.c:
```diff
@@ -185,13 +185,45 @@ static ALWAYS_INLINE void clkInit(void)
+				(CLOCK_GetDiv(kCLOCK_Usdhc1Div) + 1U);
+	} else if (nusdhc == 0) {
+		return CLOCK_GetSysPfdFreq(kCLOCK_Pfd0) /
+				(CLOCK_GetDiv(kCLOCK_Usdhc2Div) + 1U);
```
It looks like you added this logic to `clock_control_mcux_ccm.c` but forgot to remove it from here

MaureenHelm commented on soc/arm/nxp_imx/rt/soc.h:
```diff
@@ -23,6 +23,19 @@ extern "C" {
 
+#if defined(CONFIG_DISK_ACCESS_NXP_USDHC1) ||	\
+	defined(CONFIG_DISK_ACCESS_NXP_USDHC2)
+u32_t usdhc_src_clk_freq(u16_t nusdhc);
```
This can be removed

JunYangNXP replied:
fixed

JunYangNXP committed:
sdhc: rename disk_access_sdhc.c

The name disk_access_sdhc.c is ambiguous,

actually this driver depends on SPI,

rename this file.

In addition, move the generic sdhc stuff from C file

to head file for other sdhc drivers to use.

1) disk_access_sdhc.c->disk_access_spi_sdhc.c.

2) create .h and move sdhc specifications from .c to .h.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
dts/sdhc: support mmc dts and usdhc dts

1) dts/bindings/mmc/mmc.yaml: specifies common mmc.

2) dts/bindings/mmc/nxp,imx-usdhc.yaml: specifies
   nxp usdhc module which inherits mmc.

3) dts/arm/nxp/nxp_rt.dtsi: usdhc support on RT chip.

4) boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts:
   usdhc slot support on mimxrt1050_evk board.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
Kconfig: Kconfig support i.MX RT USDHC

Add Kconifg options to support i.MXRT USDHC driver.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
usdhc/pinmux: configure pinmux of usdhc on mimxrt1050 evk

Implementation of pinmux of usdhc depends on board design.

Usdhc driver could change pinmux according to SD mode, SoC

should provide API for this. Board pinmux should register

its pinmux function to SoC.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
clock/usdhc: Enable clock of USDHC of i.MXRT

Enable clock of USDHC of i.MXRT.

Add clock interface for USDHC in ccm.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
usdhc: support NXP i.MXRT usdhc

support NXP i.MXRT usdhc for file system to access.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP committed:
samples/fs: support fs demo on mimxrt_1050 evk board

Add mimxrt_1050 evk board into white list.

FS demo Kconfig for mimxrt_1050 evk board.

Signed-off-by: Jun Yang <jun.yang@nxp.com>

JunYangNXP commented on subsys/disk/CMakeLists.txt:
```diff
@@ -3,4 +3,5 @@
 zephyr_sources_ifdef(CONFIG_DISK_ACCESS_RAM disk_access_ram.c)
-zephyr_sources_ifdef(CONFIG_DISK_ACCESS_SDHC disk_access_sdhc.c)
+zephyr_sources_ifdef(CONFIG_DISK_ACCESS_SPI_SDHC disk_access_spi_sdhc.c)
+zephyr_sources_ifdef(CONFIG_DISK_ACCESS_USDHC disk_access_usdhc.c)
```
fixed

JunYangNXP commented on samples/subsys/fs/prj.conf:
```diff
@@ -2,8 +2,8 @@ CONFIG_SPI=y
-CONFIG_FAT_FILESYSTEM_ELM=y
-CONFIG_PRINTK=y
\ No newline at end of file
+CONFIG_FAT_FILESYSTEM_ELM=y
```
Fixed

MaureenHelm reviewed (approved):
@ydamigos @agansari please revisit

ydamigos commented on subsys/disk/disk_access_sdhc.h:
```diff
@@ -0,0 +1,746 @@
+/* Called at the end of a retry loop.  Returns if the minimum try
+ * count and timeout has passed.  Delays/yields on retry.
+ */
+static inline bool sdhc_retry_ok(struct sdhc_retry *retry)
```
Why did you move it to `disk_access_sdhc.h`, if it is only used in `disk_access_spi_sdhc.c`? The same questions applies to other functions in `disk_access_sdhc.h` which exist in one of the `disk_access_*.c` files?

MaureenHelm replied:
@junyangnxp please respond

JunYangNXP replied:
Actually I moved all the functions which are not associated to SPI from disk_access_spi_sdhc.c to disk_access_sdhc.h. Although some functions only are used by disk_access_spi_sdhc.c now, they are potential to be used by future other vender's SDHC drivers.

ydamigos reviewed (commented):
Tested it on olimexino_stm32 and it works.

marc-hb commented:
The sanitycheck build is broken after this merge, more specifically these two configurations:
```
$ZEPHYR_BASE/scripts/sanitycheck -T $ZEPHYR_BASE/samples/subsys/fs/ -v -p olimexino_stm32 -p nrf52840_blip

zephyr/subsys/disk/disk_access_spi_sdhc.c:707:13: error: Macro is deprecated [-Werror]
   DT_INST_0_ZEPHYR_MMC_SPI_SLOT_CS_GPIO_CONTROLLER);
             ^~~~~~~~~~~~~~~~~~~~~
zephyr/subsys/disk/disk_access_spi_sdhc.c:707:13: error: Macro is deprecated [-Werror]
zephyr/subsys/disk/disk_access_spi_sdhc.c:710:13: error: Macro is deprecated [-Werror]
  data->pin = DT_INST_0_ZEPHYR_MMC_SPI_SLOT_CS_GPIO_PIN;
             ^~~~~~~~~~~~~~~~~~~~~
zephyr/subsys/disk/disk_access_spi_sdhc.c:710:13: error: Macro is deprecated [-Werror]
```

```
include/generated/generated_dts_board_unfixed.h

#define DT_INST_0_ZEPHYR_MMC_SPI_SLOT_CS_GPIO_CONTROLLER __DEPRECATED_MACRO     DT_ST_STM32_SPI_40003800_ZEPHYR_MMC_SPI_SLOT_0_CS_GPIO_CONTR>
#define DT_INST_0_ZEPHYR_MMC_SPI_SLOT_CS_GPIO_FLAGS __DEPRECATED_MACRO          DT_ST_STM32_SPI_40003800_ZEPHYR_MMC_SPI_SLOT_0_CS_GPIO_FLAGS
#define DT_INST_0_ZEPHYR_MMC_SPI_SLOT_CS_GPIO_PIN __DEPRECATED_MACRO            DT_ST_STM32_SPI_40003800_ZEPHYR_MMC_SPI_SLOT_0_CS_GPIO_PIN
```
I bisected this to this commit:
    c7e625f2b39b jun.yang@nxp.com sdhc: rename disk_access_sdhc.c

@nashif how come CI does not seem to fail this?

BTW next time please separate the .c rename and the .h creation into two different commits. Now even `git log -M01 -C01` is completely lost, it's impossible to see what changed if anything and let's not even get into `git blame`

JunYangNXP commented:
@marc-hb My fault:( Will I create PR to fix this sanitycheck issue or someone else will take it?

BR

marc-hb commented:
> @nashif how come CI does not seem to fail this? Doesn't test these particular combinations?

Got some answers from @nashif . Long story short: CI runs different configurations based on which files have changed. So this can happen.

galak commented:
I made a fix for this in #17519
